sdk/core/azure_core_amqp/Test-Setup.ps1

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# cspell: ignore JOBID
+
+param (
+  [string]$PackageName
+)
+
+# Load common ES scripts
+. "$PSScriptRoot\..\..\..\eng\common\scripts\common.ps1"
+
+$WorkingDirectory = ([System.IO.Path]::Combine($RepoRoot, "../TestArtifacts"))
+
+if (-not $PackageName) {
+  Write-Host "PackageName parameter not provided."
+  exit 1
+}
+
+# Create the working directory if it does not exist.
+Write-Host Using Working Directory $WorkingDirectory
+
+if (-not (Test-Path $WorkingDirectory)) {
+  Write-Host "Working directory does not exist, creating working directory: $WorkingDirectory"
+  New-Item -ItemType Directory -Path $WorkingDirectory
+}
+
+Write-Host "Setting current directory to working directory: $WorkingDirectory"
+Push-Location -Path $WorkingDirectory
+
+# Clone and build the Test Amqp Broker.
+try {
+
+  $repositoryUrl = "https://github.com/Azure/azure-amqp.git"
+  $cloneCommand = "git clone $repositoryUrl"
+
+  Write-Host "Cloning repository from $repositoryUrl..."
+  Invoke-LoggedCommand $cloneCommand
+
+  Set-Location -Path "./azure-amqp/test/TestAmqpBroker"
+  Invoke-LoggedCommand "dotnet restore"
+  if (!$? -ne 0) {
+    Write-Error "Failed to restore dependencies for TestAmqpBroker."
+    exit 1
+  }
+
+  Invoke-LoggedCommand "dotnet build"
+  if (!$? -ne 0) {
+    Write-Error "Failed to build TestAmqpBroker."
+    exit 1
+  }
+
+  Invoke-LoggedCommand "dotnet publish --self-contained --framework net6.0"
+
+  Write-Host "Test broker built successfully."
+
+  # now that the Test broker has been built, launch the broker on a local address.
+  $env:TEST_BROKER_ADDRESS = 'amqp://127.0.0.1:25672'
+
+  Write-Host "Starting test broker listening on " $env:TEST_BROKER_ADDRESS "..."
+
+  if ($IsLinux) {
+    Set-Location -Path $WorkingDirectory/azure-amqp/bin/Debug/TestAmqpBroker/net6.0/linux-x64/publish
+  }
+  elseif ($IsMacOS) {
```

Comment:
My Mac is an M4, so this won't work. But I don't think we should do it this way anyway. That wasn't my intent with `--self-contained` when I suggested it. That was supposed to be in the Azure/azure-amqp repo and you pull them / or only what you need down.

Still, if you can correctly publish packages, you're making *way* more work than you need to not to mention leaving stuff all over the drive. Did you even `.gitignore` these? You shouldn't be putting build artifacts just anywhere in the repo, even if they are ignored, unless you're collapsing to a single directory tree and avoiding many copies.

Reply:
PowerShell runs on M4 just fine. I do it all the time.

You say it's not supposed to run on a dev machine now. When I first wrote the test resources scripts, they also weren't designed to run as part of the CI. Things change. But I didn't back myself into a corner then so the change was minimal.

The reason I recommended `dotnet publish` was to package the runtime to make it available for download. It's an unnecessary burden for *every* agent to do it when using it vs. a few platform agents doing it once when the Azure/azure-amqp repo does a new release.

Reply:
Oh, and for the M4 thingy: The core problem with this is that dotnet publish, the output binaries are put in CPU specific output directories (x86, x64, and arm64). I need to find those directories so that I can CD into them to find the executable to run.

And I don't know how to find the host CPU architecture in Powershell - there doesn't appear to be an $IsX86, $IsArm or $IsX64 directory.

Reply:
I started and email thread with .NET and CC'd you. `dotnet <assembly>` should work and is easier. We just need to figure out why it's not working on macOS.